### PR TITLE
Simplify thrift mime types

### PR DESCRIPTION
**Pull request by ajaygeorge**
Simplify thrift mime types to avoid the usage of parameters.
This is inspired from how grpc uses mime types.
e.g. The old mime type application/x-thrift; t=binary will be replaced
with application/x-thrift+binary.

### CODE REVIEW

**Comment by tdcmeehan** [comment]
Idea sounds good, is the motivation just to simplify the code?

**Comment by ajaygeorge** [comment]
> Idea sounds good, is the motivation just to simplify the code?

I wanted to keep our implementation aligned with a standard instead of custom mime types.
The closest I could think of was `grpc` which uses this format.

```
Content-Type → "content-type" "application/grpc" [("+proto" / "+json" / {custom})]
```
from : https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#requests

File: http-client/src/main/java/com/facebook/airlift/http/client/thrift/ThriftRequestUtils.java [hunk 1]
```diff
@@ -25,10 +27,10 @@
 
 public class ThriftRequestUtils
 {
-    private static final MediaType THRIFT_TYPE = MediaType.create("application", "x-thrift");
-    private static final String TYPE_BINARY = THRIFT_TYPE.withParameter("t", Protocol.BINARY.name()).toString().toLowerCase();
-    private static final String TYPE_COMPACT = THRIFT_TYPE.withParameter("t", Protocol.COMPACT.name()).toString().toLowerCase();
-    private static final String TYPE_FBCOMPACT = THRIFT_TYPE.withParameter("t", Protocol.FB_COMPACT.name()).toString().toLowerCase();
+    public static final String TYPE_BINARY = "application/x-thrift+binary";
```
Comment (tdcmeehan) [inline http-client/src/main/java/com/facebook/airlift/http/client/thrift/ThriftRequestUtils.java:30]: If it's easy, I'd create an enum for this.

File: http-client/src/main/java/com/facebook/airlift/http/client/thrift/ThriftResponseHandler.java [hunk 2]
```diff
@@ -49,6 +53,23 @@ public ThriftResponseHandler(ThriftCodec<T> thriftCodec)
     @Override
     public ThriftResponse<T> handle(Request request, Response response)
     {
+        if (isClientError(response.getStatusCode())) {
+            ByteSource byteSource = new ByteSource() {
```
Comment (tdcmeehan) [inline http-client/src/main/java/com/facebook/airlift/http/client/thrift/ThriftResponseHandler.java:57]: For readability I'd factor this into a private method.

File: jaxrs/src/main/java/com/facebook/airlift/jaxrs/thrift/ThriftMapper.java [hunk 3]
```diff
@@ -72,7 +75,7 @@ public Object readFrom(
         }
         catch (Exception e) {
             // we want to return a 400 for bad Thrift but not for a real IO exception
-            if (e instanceof IOException && !(e instanceof EOFException)) {
+            if (e instanceof IOException && !(e instanceof ThriftProtocolException) && !(e instanceof EOFException)) {
```
Comment (tdcmeehan) [inline jaxrs/src/main/java/com/facebook/airlift/jaxrs/thrift/ThriftMapper.java:78]: Should this be in a separate commit?  Doesn't seem related?

**Review by tdcmeehan — COMMENTED** [review]
Minor nits

File: http-client/src/main/java/com/facebook/airlift/http/client/thrift/ThriftRequestUtils.java (hunk 1 above)
Comment (ajaygeorge) [reply http-client/src/main/java/com/facebook/airlift/http/client/thrift/ThriftRequestUtils.java:30]: `@Consumes` and `@Produces` annotations takes in a string. With an enum we will need to have an additional `.name()` invocation.
Also the `javax.ws.rs.core.MediaType` class which I have used as a reference uses strings as the mime types.

File: http-client/src/main/java/com/facebook/airlift/http/client/thrift/ThriftResponseHandler.java (hunk 2 above)
Comment (ajaygeorge) [reply http-client/src/main/java/com/facebook/airlift/http/client/thrift/ThriftResponseHandler.java]: done

File: jaxrs/src/main/java/com/facebook/airlift/jaxrs/thrift/ThriftMapper.java (hunk 3 above)
Comment (ajaygeorge) [reply jaxrs/src/main/java/com/facebook/airlift/jaxrs/thrift/ThriftMapper.java:78]: Done. splitted into two commits

**Review by tdcmeehan — APPROVED** [review]
